chains/evm/contracts/test/onRamp/OnRamp/OnRamp.getCCVsForPool.t.sol

```diff
@@ -0,0 +1,165 @@
+
+    assertEq(result.length, expectedCCVs.length, "Should surface pool-provided CCVs");
+    assertEq(result[0], expectedCCVs[0], "First CCV should match");
+    assertEq(result[1], expectedCCVs[1], "Second CCV should match");
+  }
+
+  function test_getCCVsForPool_ReturnsDefaultCCVs_WhenPoolDoesNotSupportV2() public {
+    address[] memory expectedCCVs = new address[](1);
+    expectedCCVs[0] = makeAddr("poolCCV");
+    address pool = _deployPoolV2(expectedCCVs);
+
+    // Force the ERC165 probe to report that IPoolV2 is unsupported.
+    vm.mockCall(
+      pool, abi.encodeWithSelector(IERC165.supportsInterface.selector, type(IPoolV2).interfaceId), abi.encode(false)
+    );
+
+    address[] memory result = s_onRampTestHelper.getCCVsForPool(DEST_CHAIN_SELECTOR, s_token, 100, 0, "");
+
+    assertEq(result.length, 1, "Should fall back to default CCV when pool is V1");
+    assertEq(result[0], s_helperDefaultCCV, "Returned CCV should be the helper default");
+  }
+
+  function test_getCCVsForPool_ReturnsDefaultCCVs_WhenPoolReturnsEmptyArray() public {
+    _deployPoolV2(new address[](0));
+
+    address[] memory result = s_onRampTestHelper.getCCVsForPool(DEST_CHAIN_SELECTOR, s_token, 100, 0, "");
+
+    assertEq(result.length, 1, "Should fall back to default CCV when pool is silent");
+    assertEq(result[0], s_helperDefaultCCV, "Returned CCV should be the helper default");
+  }
+
+  function test_getCCVsForPool_PassesThroughAddressZeroSentinel() public {
+    address[] memory poolCCVs = new address[](3);
+    poolCCVs[0] = makeAddr("poolCCV1");
+    poolCCVs[1] = address(0);
+    poolCCVs[2] = makeAddr("poolCCV2");
+
+    _deployPoolV2(poolCCVs);
+
+    address[] memory result = s_onRampTestHelper.getCCVsForPool(DEST_CHAIN_SELECTOR, s_token, 100, 0, "");
+
+    assertEq(result.length, poolCCVs.length, "Should preserve address(0) sentinel for merge step");
+    assertEq(result[0], poolCCVs[0], "First CCV should match");
+    assertEq(result[1], address(0), "Sentinel should be present");
+    assertEq(result[2], poolCCVs[2], "Third CCV should match");
+  }
+
+  function _configureHelperDestChain() internal {
+    address[] memory defaultCCVs = new address[](1);
+    defaultCCVs[0] = makeAddr("helperDefaultCCV");
+    s_helperDefaultCCV = defaultCCVs[0];
+
+    OnRamp.DestChainConfigArgs[] memory destChainConfigArgs = new OnRamp.DestChainConfigArgs[](1);
+    destChainConfigArgs[0] = OnRamp.DestChainConfigArgs({
+      destChainSelector: DEST_CHAIN_SELECTOR,
+      router: s_sourceRouter,
+      laneMandatedCCVs: new address[](0),
+      defaultCCVs: defaultCCVs,
+      defaultExecutor: makeAddr("helperExecutor"),
+      offRamp: abi.encodePacked(address(s_offRampOnRemoteChain))
+    });
+
+    s_onRampTestHelper.applyDestChainConfigUpdates(destChainConfigArgs);
+  }
+
+  function _deployPoolV2(
+    address[] memory requiredCCVs
+  ) internal returns (address pool) {
+    pool = address(new MockPoolV2(requiredCCVs));
```

Comment:
Why do we have a MockPool that really uses 165 but then also mock all calls involving 165?